app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,70 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[ show edit update destroy ]
+
+  # GET /recipes or /recipes.json
+  def index
+    @recipes = Recipe.all
+  end
+
+  # GET /recipes/1 or /recipes/1.json
+  def show
+  end
+
+  # GET /recipes/new
+  def new
+    @recipe = Recipe.new
+  end
+
+  # GET /recipes/1/edit
+  def edit
+  end
```

Comment:
Great work with the actions for recipes. The requirements ask to implement all actions except for update. Please remove the edit action.